### PR TITLE
Added utility functions to build AST's from command/function inputs, new API's exposing LiveRunner and Mirror functionality to external host applications

### PR DESCRIPTION
**Pull request by aparajit-pratap**
New utility API to build AST in ProtoCore/ASTCompilerUtils.cs, additional API's in LiveRunner for use by external host: UpdateGraph(AssociativeNode astNode), List<LibraryMirror> ResetVMAndImportLibrary(List<string> libraries)

### CODE REVIEW

**Comment by lukechurch** [comment]
Hey Aparajit, thanks for this - I need to know what the test coverage on the code being refactored is, and we'll need new tests on the new APIs before acceptance

**Comment by aparajit-pratap** [comment]
Hi Luke, these api's are all being consumed only by DS1 right now and are tested only through the prototypes we have been building for demos. The ones in LiveRunner such as UpdateGraph(AssociativeNode astNode), List ResetVMAndImportLibrary(List libraries) are merely wrappers of existing API's. I can add Nunit tests for these and the ones in ASTCompilerUtils.cs if you wish.

File: Core/ProtoScript/Runners/LiveRunner.cs [hunk 1]
```diff
@@ -477,6 +479,25 @@ public void UpdateGraph(GraphSyncData syncData)
         }
 
         /// <summary>
+        /// Called for delta execution of AST node input
+        /// </summary>
+        /// <param name="astNode"></param>
+        public void UpdateGraph(AssociativeNode astNode)
```
Comment (lukechurch) [inline Core/ProtoScript/Runners/LiveRunner.cs:486]: Given that this does nothing if it isn't a code block node, why not specify it as one? Is the idea that it's a partial implementation that will become more general? If so it should throw a not implemented.

File: Core/ProtoScript/Runners/LiveRunner.cs [hunk 2]
```diff
@@ -839,6 +860,52 @@ public void ResetVMAndResyncGraph(List<string> libraries)
             UpdateCmdLineInterpreter(code);
         }
 
+        /// <summary>
+        /// Used temporarily to reset the VM whenever a new library is imported and re-executes 
+        /// Returns the list of new Library Mirrors for reflection
```
Comment (lukechurch) [inline Core/ProtoScript/Runners/LiveRunner.cs:880]: Descriptions of methods should state what the method does, not where it is used.

If this is only a temporary tool, it shouldn't be in LiveRunner which is a production API

**Comment by lukechurch** [comment]
Understood.

So - anything that goes in LiveRunner needs to have 90%+ test coverage whether it's a wrapper or not. I don't think we're there at the moment, but we need to start improving to iterate towards that.

The utility methods are likely to get folded into the core ASTbuilder pretty quickly and therefore need test coverage.

The additional methods that are only used by the DS1 prototype are ok without test coverage for now, esp. if it's a pain to do in C++.

Thanks :)

**Comment by aparajit-pratap** [comment]
Added test cases and comments as required.

**Pull request merged**